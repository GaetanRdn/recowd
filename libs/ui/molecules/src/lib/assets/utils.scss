@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
