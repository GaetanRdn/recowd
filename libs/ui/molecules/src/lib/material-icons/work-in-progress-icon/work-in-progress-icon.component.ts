import { ChangeDetectionStrategy, Component } from '@angular/core';
import { MaterialIconDirective } from '@recowd/ui-atoms';

@Component({
  selector: 'rc-work-in-progress-icon',
  standalone: true,
  imports: [MaterialIconDirective],
  template: ` <rc-material-icon type="outlined">engineering</rc-material-icon>`,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class WorkInProgressIconComponent {}
