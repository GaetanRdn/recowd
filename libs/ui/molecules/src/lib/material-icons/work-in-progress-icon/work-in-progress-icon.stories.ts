import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { WorkInProgressIconComponent } from './work-in-progress-icon.component';

export default {
  title: 'molecules/material-icons/work-in-progress-icon',
  parameters: {
    jest: 'work-in-progress-icon',
  },
  decorators: [
    moduleMetadata({
      imports: [WorkInProgressIconComponent],
    }),
  ],
} as Meta<WorkInProgressIconComponent>;

const Template: Story<WorkInProgressIconComponent> = () => ({
  template: `<rc-work-in-progress-icon></rc-work-in-progress-icon>`,
});

export const Default = Template.bind({});
