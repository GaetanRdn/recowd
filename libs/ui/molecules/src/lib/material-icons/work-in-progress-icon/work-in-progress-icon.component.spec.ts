import { TestBed, waitForAsync } from '@angular/core/testing';

import { WorkInProgressIconComponent } from './work-in-progress-icon.component';
import { TemplateLookup } from '@recowd/test/utils';

describe('WorkInProgressIconComponent', () => {
  let templateLookup: TemplateLookup<WorkInProgressIconComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [WorkInProgressIconComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(WorkInProgressIconComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.fixture.nativeElement.children).toMatchSnapshot());
});
