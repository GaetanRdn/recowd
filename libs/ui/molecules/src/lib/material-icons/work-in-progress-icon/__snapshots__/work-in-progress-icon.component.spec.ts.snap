// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkInProgressIconComponent should create 1`] = `
HTMLCollection [
  <rc-material-icon
    class="material-icons-outlined"
    type="outlined"
  >
    engineering
  </rc-material-icon>,
]
`;
